repl.rkt

```diff
@@ -0,0 +1,82 @@
+#lang racket/gui
+#|
+NOTICE: modify from https://github.com/racket-templates/guiapp/blob/master/main.rkt based on MIT
+origin author: Stephen De Gabrielle(GitHub: @spdegabrielle)
+modifier author: Lîm Tsú-thuàn(GitHub: @dannypsnl)
+|#
+
+(provide repl-text%)
+
+(require framework)
+
+(define repl-text%
+  ;;; TODO: should share some operations with code editor
+  (class racket:text%
+    (inherit insert last-position get-text erase)
+    (define prompt-pos 0)
+    (define locked? #f)
+    ;;; these two prevent users accidentally remove `> ` or typed commands
+    (define/augment can-insert?
+      (lambda (start len)
+        (and (>= start prompt-pos) (not locked?))))
+    (define/augment can-delete?
+      (lambda (start end)
+        (and (>= start prompt-pos) (not locked?))))
+    ;;; hajack special keys
+    (define/override (on-char c)
+      (super on-char c)
+      (match (send c get-key-code)
+        [#\return
+         (when (not locked?)
+           (set! locked? #t)
+           (define result
+             (eval (read (open-input-string (get-text prompt-pos (last-position))))))
+           (output (format "~a~n" result))
+           (new-prompt))]
+        [else (void)]))
+    ;; methods
+    (define/public (new-prompt)
+      (queue-output (lambda ()
+                      (set! locked? #f)
+                      (insert "> ")
+                      (set! prompt-pos (last-position)))))
+    (define/public (run-file filepath)
+      (queue-output (lambda ()
+                      (let ((was-locked? locked?))
+                        (set! locked? #f)
+                        (insert (eval (read (open-input-file filepath))))
+                        (set! locked? was-locked?)))))
```

Comment:
TODO: find the correct way to evaluate the whole file and print output

Reply:
Suddenly realize it should attach to process stdout/stderr port to work with the application

Reply:
https://docs.racket-lang.org/reference/eval.html#%28def._%28%28quote._~23~25kernel%29._load%29%29 Maybe this?

Reply:
Use `dynamic-enter!` for now.